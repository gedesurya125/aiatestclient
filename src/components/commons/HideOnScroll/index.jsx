import React from 'react'
import useScrollTrigger from '@mui/material/useScrollTrigger';
import Slide from '@mui/material/Slide';

const HideOnScroll = ({children}) => {
  const trigger = useScrollTrigger();
  return (
    <Slide data-testid="hide-on-scroll" appear={false} direction="down" in={!trigger}>
      {children}
    </Slide>
  )
}

export default HideOnScroll
