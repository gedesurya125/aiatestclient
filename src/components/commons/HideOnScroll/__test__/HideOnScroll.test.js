import React from "react";
import ReactDOM from "react-dom";
import HideOnScroll from "../index";
import {render, cleanup} from '@testing-library/react';
import renderer from 'react-test-renderer';


afterEach(cleanup);
it("render without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<HideOnScroll><p>Some Text</p></HideOnScroll>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("render HideOnScroll component correctly", () => {
  const {getByTestId} = render(<HideOnScroll><p>Some Text</p></HideOnScroll>);
  expect(getByTestId('hide-on-scroll')).toContainHTML('<p data-testid="hide-on-scroll">Some Text</p>');
})

it("render HideOnScroll component correctly with text", () => {
  const {getByTestId} = render(<HideOnScroll><p></p></HideOnScroll>);
  expect(getByTestId('hide-on-scroll')).not.toHaveAttribute();
})

it("Matches snapshot 1", () => {
  const tree = renderer.create(<HideOnScroll><p></p></HideOnScroll>).toJSON();
  expect(tree).toMatchSnapshot();
})

it("Matches snapshot 2", () => {
  const tree = renderer.create(<HideOnScroll><div></div></HideOnScroll>).toJSON();
  expect(tree).toMatchSnapshot();
})


