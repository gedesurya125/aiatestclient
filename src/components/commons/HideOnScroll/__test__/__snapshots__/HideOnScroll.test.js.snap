// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1 1`] = `
<p
  data-testid="hide-on-scroll"
  style={
    Object {
      "visibility": undefined,
    }
  }
/>
`;

exports[`Matches snapshot 2 1`] = `
<div
  data-testid="hide-on-scroll"
  style={
    Object {
      "visibility": undefined,
    }
  }
/>
`;
